### PR TITLE
TST: Blacklist pytest 3.7

### PR DESCRIPTION
**Pull request by sbanert**
This pull request blacklists those `pytest` versions which cause an infinite recursion (see the discussion in #1416). The `pytest` issue was fixed here https://github.com/pytest-dev/pytest/pull/3771 and should be included in version `3.7.2`.

### CODE REVIEW

**Comment by kohr-h** [comment]
> have again no idea what’s going on here. This time, it’s definitely a non-blacklisted version of pytest (3.7.2), and still some errors occur in odl.util.

The weird thing is that only `odl.util` is affected. Without taking a detailed look, I suspect the quick fix for the infinite recursion bug since it mocks around with the collection of tests. The error occurs when `getattr(module, "pytest_plugins", [])` returns a module instead of a list (in our case, `module` is the `odl.util` module).
For now I agree, blacklist all of the 3.7 releases, but if possible, file a bug report on the pytest issue tracker. I'm sure there are not so many with the same setup as ours, so it might really be a corner case.

File: odl/util/testutils.py [hunk 1]
```diff
@@ -648,6 +648,11 @@ def test(arguments=None):
         raise ImportError('ODL tests cannot be run without `pytest` installed.'
                           '\nRun `$ pip install [--user] odl[testing]` in '
                           'order to install `pytest`.')
+    if pytest.__version__ in ['3.7.0', '3.7.1']:
+        raise ImportError('ODL tests cannot be run with `pytest` '
```
Comment (kohr-h) [inline odl/util/testutils.py]: Make this a `RuntimeError`

File: odl/util/testutils.py [hunk 2]
```diff
@@ -648,6 +648,11 @@ def test(arguments=None):
         raise ImportError('ODL tests cannot be run without `pytest` installed.'
                           '\nRun `$ pip install [--user] odl[testing]` in '
                           'order to install `pytest`.')
+    if pytest.__version__ in ['3.7.0', '3.7.1']:
```
Comment (kohr-h) [inline odl/util/testutils.py]: This doesn't cover the same range as the pip requirements (not 3.7.2 and not the possibly broken subsequent versions).
Suggestion:
```python
from packaging import version
...
# TODO(sbanert): adapt constraint when upstream issue is fixed
# See <link to pytest issue>
if not version.parse(pytest.__version__) < version.parse("3.7*"):
    raise RuntimeError(...)
```
This uses the faster-to-import `packaging.version`, and it includes dev versions like 3.7.0dev0.

File: odl/util/testutils.py [hunk 3]
```diff
@@ -648,6 +648,11 @@ def test(arguments=None):
...
+        raise ImportError('ODL tests cannot be run with `pytest` '
+                          'versions `3.7.0` and `3.7.1`.\nRun `$ pip install '
+                          '[--user] odl[testing]` in order to install '
+                          '`pytest`.')
```
Comment (kohr-h) [inline odl/util/testutils.py]: The helpful part of the error message isn't that helpful here, better `Run pip install "pytest<3.7"`

File: odl/util/testutils.py (hunk 2 above)
Comment (kohr-h) [reply odl/util/testutils.py]: I mean faster than `pkg_resources.parse_version`. The dependency on `packaging` has been added recently.
